src/theory/quantifiers/sygus/sygus_interpol.cpp

```diff
@@ -40,25 +40,206 @@ SygusInterpol::SygusInterpol(LogicInfo logic) : d_logic(logic) {}
+      d_vlvsShared.push_back(vlv);
+      d_varTypesShared.push_back(tn);
+    }
+  }
+  // make the sygus variable list
+  d_ibvlShared = nm->mkNode(kind::BOUND_VAR_LIST, d_vlvsShared);
+  Trace("sygus-interpol-debug") << "...finish" << std::endl;
 }
 
 std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> > getIncludeCons(
     const std::vector<Node>& assumptions, const Node& conclusion)
 {
+  NodeManager* nm = NodeManager::currentNM();
+  Assert(options::produceInterpols() != options::ProduceInterpols::NONE);
   std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> > result =
       std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >();
+  // ASSUMPTIONS
+  if (options::produceInterpols() == options::ProduceInterpols::ASSUMPTIONS)
+  {
+    Node tmpAssumptions;
+    if (assumptions.size() == 1)
+    {
+      tmpAssumptions = assumptions[0];
+    }
+    else
+    {
+      tmpAssumptions = nm->mkNode(kind::AND, assumptions);
+    }
+    expr::getOperatorsMap(tmpAssumptions, result);
+  }
+  // CONJECTURE
+  else if (options::produceInterpols() == options::ProduceInterpols::CONJECTURE)
+  {
+    expr::getOperatorsMap(conclusion, result);
+  }
+  // SHARED
+  else if (options::produceInterpols() == options::ProduceInterpols::SHARED)
+  {
+    // Get operators from assumptions
+    std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >
+        include_cons_assumptions =
+            std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >();
+    Node tmpAssumptions;
+    if (assumptions.size() == 1)
+    {
+      tmpAssumptions = assumptions[0];
+    }
+    else
+    {
+      tmpAssumptions = nm->mkNode(kind::AND, assumptions);
+    }
+    expr::getOperatorsMap(tmpAssumptions, include_cons_assumptions);
+
+    // Get operators from conclusion
+    std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >
+        include_cons_conclusion =
+            std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >();
+    expr::getOperatorsMap(conclusion, include_cons_conclusion);
+
+    // Compute intersection
+    for (std::map<TypeNode,
+                  std::unordered_set<Node, NodeHashFunction> >::iterator itec =
+             include_cons_assumptions.begin();
+         itec != include_cons_assumptions.end();
+         itec++)
+    {
+      TypeNode tn = itec->first;
+      std::unordered_set<Node, NodeHashFunction> assumptionsOps = itec->second;
+      std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >::iterator
+          concIter = include_cons_conclusion.find(tn);
+      if (concIter != include_cons_conclusion.end())
+      {
+        std::unordered_set<Node, NodeHashFunction> conclusionOps =
+            concIter->second;
+        std::unordered_set<Node, NodeHashFunction> conclusionOpsSet =
+            std::unordered_set<Node, NodeHashFunction>(conclusionOps.begin(),
+                                                       conclusionOps.end());
```

Comment:
```suggestion
        std::unordered_set<Node, NodeHashFunction> conclusionOpsSet{
            conclusionOps.begin(), conclusionOps.end()};
```
But more generally: why is this created?